tutorials/plot_stats.py

```diff
@@ -11,40 +11,126 @@
 
 ###################################################################################################
-# Subtitle
+# 1. Compute and plot different shuffles of spikes
 # ~~~~~~~~
 #
-# Words, words, words.
+# Use spikes_ms to compute shuffle_isis (with 10 shuffles), shuffle_bins (with 10 shuffles), 
+# shuffle_poisson (with 10 shuffles), and shuffle_circular (with 10 shuffles).
+# Also compute a vector permutation, for comparison.
 #
 
 ###################################################################################################
 
-# Do some code stuff
-print('Hello World!')
+# Shuffle spike ms using four methods
+shuffled_isis = shuffle_isis(spikes_ms, n_shuffles=10)
+shuffled_bins = shuffle_bins(spikes_ms, bin_width_range=[50, 200], n_shuffles=10)
+shuffled_poisson = shuffle_poisson(spikes_ms, n_shuffles=10)
+shuffled_circular = shuffle_circular(spikes_ms, shuffle_min=200, n_shuffles=10)
+
+# Do a vector permutation, for comparison
+perm_spikes = vec_perm(spikes_ms, n_perms=10)
 
 ###################################################################################################
 
-# Do more code stuff
-print('More stuff!')
+# Plot non-shuffled, all different shuffles, and vector permutation
+f, ((ax1, ax2, ax3), (ax4, ax5, ax6)) = plt.subplots(2, 3, sharey = True)
+# original
+ax1.eventplot(spikes_ms[0:50])
+ax1.set_title('Non-shuffled')
+ax1.set_xlim([0, 6000])
+ax1.set_xlabel('time')
+
+# isis
+ax2.eventplot(shuffled_isis[:, 0:80])
+ax2.set_title('Shuffle ISIS n_shuffles = 10')
+ax2.set_xlim([0, 6000])
+ax2.set_xlabel('time')
+
+# poisson
+ax3.eventplot(shuffled_poisson[:, :])
+ax3.set_title('Shuffle poisson n_shuffles = 10')
+ax3.set_xlim([0, 6000])
+ax3.set_xlabel('time')
+
+# only plot up to 55th so the effect is visible
+ax4.eventplot(perm_spikes[:, :55])
+ax4.set_title('Permutated spikes n_permutations = 10')
+ax4.set_xlim([0, 6000])
+ax4.set_xlabel('time')
+
+# shuffled bins
+ax5.eventplot(shuffled_bins[:, :])
+ax5.set_title('Shuffle bins n_shuffles = 10')
+ax5.set_xlim([0, 6000])
+ax5.set_xlabel('time')
+
+# shuffled circular
+ax6.eventplot(shuffled_circular[:, :])
+ax6.set_title('Shuffle circular n_shuffles = 10')
+ax6.set_xlim([0, 6000])
+ax6.set_xlabel('time')
+
+# Add some padding between subplots
+# f.tight_layout(h_pad = 0.001)
+# Make figure bigger
+f.set_size_inches((40/2.54, 20/2.54))
+plt.subplots_adjust(left=None, bottom=None, right=None, top=None, wspace=None, hspace=0.3)
 
 ###################################################################################################
-# Conclusion
-# ~~~~~~~~~~
+# 2. Compute the empirical p-value from distribution of surrogates
+# ~~~~~~~~
 #
-# Words, words, words.
+# Compute the empirical p-value of 100 from the distribution of surrogates (spikes_ms).
 #
+
+###################################################################################################
+
+# Compute the empirical p-value of 100
+compute_empirical_pvalue(100, spikes_ms)
```

Comment:
This is quite the usage of `compute_empirical_pvalue` - the idea here would be that the second input of the function should be a distribution of computed test values on the surrogates (for example, a vector of t-values), and the first input should be the same test on the real data. 

We will have to think a little on how to best set this up in a simple simulation / tutorials. Probably the easiest is to set up a firing rate change (simulate a neuron with firing rate of say 2 Hz before some "event" time, and 8 Hz after). Then, we can compare the firing rate difference of the real data, shuffle the data a bunch, and compare to surrogates. If you're not sure how to set this up - let's chat a bit more about it soon!